src/Build/Logging/FancyLogger/FancyLoggerBuffer.cs

```diff
@@ -220,19 +220,24 @@ public static FancyLoggerBufferLine WriteNewLine(FancyLoggerBufferLine line)
         public static FancyLoggerBufferLine? WriteNewLineAfterIndex(FancyLoggerBufferLine line, int lineIndex)
         {
             if (lineIndex == -1) return null;
+            // Get Id of currentTopLineIndex
```

Comment:
Few things I'm confused by in this function:
If you want to insert at the top (index 0), you'd have to pass in -1 as the index to insert after, but that means to return null. Is this a limitation?
It looks like you don't modify line before returning it. Should this just return a bool (success/failure)?
Why are we scrolling to the top line? It doesn't make more sense to scroll to the added line?

Reply:
I also thought it would make more sense to scroll to the newly added line. However, there might be cases where new lines are added outside of the current viewport and scrolling to them would mean having the viewport jumping to random lines which might not be great for the customer. 
Having it scroll to the currenTopLineIndex (the line that is currently at the top of the viewport), aka updating the display, makes more sense as it does not affect the current viewport.

Reply:
For the insertion at the top index, I agree I didn't take that into consideration. I would also like to modify several implementation details about adding and updating lines (to both make it more understandable and predictable, and also make it much more performant)